### DISCUSSION TITLE
Help understanding -X output

### DISCUSSION THREAD

**Discussion by bitdriftr** [espeak-ng/espeak-ng#1622] [Q&A] [answered]
Hello,

I am programatically parsing the output of -X. I want to match the input string with the output phoneme rules (continuation of #1614). In most cases I am able to parse what I want, but sometimes I don't understand the output.

I don't understand precisely why and when multiple "Translate 'xxxx'" blocks are outputed.

Here is multiple scenarios, I've written questions inside :

* the simple case, with a single block, easy to parse :
```
$ espeak -v fr -X bonjour
Translate 'bonjour'
  1	b        [b]

 42	on (K    [O~]
  1	o        [O]
 33	o (CC    [O]

  1	j        [Z]

 22	ou       [u]
  1	o        [O]

  1	r        [r]

 bO~Z'ur
```

* a more complex case, still possible to parse :
```
$ espeak -X syllable
Translate 'syllable'
  1	s        [s]

 20	C) y     [I]
  1	y        [j]
 20	y (C     [I]
 33	y (CC    [I]
 26	_s) y    [I]
 23	XC) y    [aI]
 24	_C) y    [aI]

  1	l        [l]
 20	l (C     [l]
 43	y) ll    [l]

118	&) able (_S4eiv  [@b@L]
 78	able     [eIb@L]
  1	a        [a]

Translate 'syll' # <------ QUESTION 1 : why is this block outputed, and how to properly "merge" it with previous ?
  1	s        [s]

 20	C) y     [I]
  1	y        [j]
 20	y (C     [I]
 33	y (CC    [I]
 26	_s) y    [I]
 23	XC) y    [aI]
 24	_C) y    [aI]

  1	l        [l]
 20	l (C     [l]
 43	y) ll    [l]

 s'Il@b@L
```

* a weird case I can't parse at all :
```
$ espeak -v fr -X arthur
Translate 'arthur'
  1	a        [a]

  1	r        [r]

 42	th (A    [_^_] # <------- QUESTION 2 : what is this phoneme ?
  1	t        [t]
 22	th       [t]

Translate 'arthur' # <----- QUESTION 3 : why does the french translation stops here and starts again in english ?
 36	ar       [A@]
  1	a        [a]

 36	th       [T]
  1	t        [t]

 97	&h) ur (_ [3]
 57	ur (_    [3:]
  1	u        [V]

(en) 'A@T3  -> (fr) 'arTWr # <-------- QUESTION 4 : why and how does this happen ?

'arTWr # <-------- QUESTION 5 : how do I map this output the the input string?
```

Thank you very much for any help

**Comment by jaacoppi** [comment, marked as answer]
Answers are in docs/dictionary.md
Question 1: See S<number>
Questions 2: See `_^_<language code>. Default value is hardcoded to English at the moment

Questions 3-5: I can't reproduce your output with version 1.51.1 or 1.52-dev. Probably related to Question #2.

My output:
$ espeak -v fr -X arthur
Translate 'arthur'
  1	a        [a]

  1	r        [r]

 42	th (A    [_^_]
  1	t        [t]
 22	th       [t]
Translate 'arthur'
 36	ar       [A@]
  1	a        [a]

 36	th       [T]
  1	t        [t]

 97	&h) ur (_ [3]
 57	ur (_    [3:]
  1	u        [V]

***@***.***(fr)

**Reply by bitdriftr** [reply]
Thank you this helped. I missed this documentation.

**Reply by anaf123** [reply]
Hi @arcln, did you manage to figure out how to map the IPA to the phonemes? If so I would appreciate some help with that as I have a similar project! :)
